src/formatting.rs

```diff
@@ -7,12 +7,12 @@ use syntect::parsing::SyntaxSet;
 use syntect::util::LinesWithEndings;
 use termcolor::WriteColor;
 
-use crate::{buffer::Buffer, cli::Theme};
+use crate::{buffer::Buffer, cli::Theme, printer::FormatOptions};
 
-pub fn get_json_formatter() -> jsonxf::Formatter {
+pub fn get_json_formatter(format_options: &FormatOptions) -> jsonxf::Formatter {
     let mut fmt = jsonxf::Formatter::pretty_printer();
-    fmt.indent = String::from("    ");
-    fmt.record_separator = String::from("\n\n");
+    fmt.indent = String::from(" ".repeat(format_options.json_indent));
+    fmt.record_separator = format_options.json_record_separator.clone();
```

Comment:
I don't think HTTPie allows customising the record separator. Can you open an issue before implementing this change?

Reply:
Ah, I see.
Since the goal is to implement all features httpie does, wouldn't it make sense to also add new features now that xh is almost fully compatible with httpie?
I only added the record separator option to be able to test whether the parser works properly though, so I don't mind dropping it.

Reply:
I feel like this option might complicate defaulting to [streamed responses](https://github.com/ducaale/xh/pull/233#issuecomment-1052445527) in the future.

Let's drop it for now and we can think about adding it in the feature.